glean-core/ios/Glean/Metrics/UuidMetric.swift

```diff
@@ -0,0 +1,104 @@
+///
+/// Instances of this class type are automatically generated by the parsers at build time,
+/// allowing developers to record values that were previously registered in the metrics.yaml file.
+///
+/// The UUID API only exposes the `UuidMetricType.generateAndSet` and `UuidMetricType.set` methods,
+/// which takes care of validating the input data and making sure that limits are enforced.
+public class UuidMetricType {
+    let handle: UInt64
+    let disabled: Bool
+    let sendInPings: [String]
+
+    /// The public constructor used by automatically generated metrics.
+    public init(category: String, name: String, sendInPings: [String], lifetime: Lifetime, disabled: Bool) {
+        self.disabled = disabled
+        self.sendInPings = sendInPings
+        self.handle = withArrayOfCStrings(sendInPings) { pingArray in
+            glean_new_uuid_metric(
+                category,
+                name,
+                pingArray,
+                Int32(sendInPings.count),
+                lifetime.rawValue,
+                disabled.toByte()
+            )
+        }
+    }
+
+    /// Destroy this metric.
+    deinit {
+        if self.handle != 0 {
+            glean_destroy_uuid_metric(self.handle)
+        }
+    }
+
+    /// Generate a new UUID and set it in the metric store.
+    ///
+    /// - returns: The `UUID` that was generated or nil if disabled
+    public func generateAndSet() -> UUID? {
+        guard !self.disabled else { return nil }
+
+        let uuid = UUID()
+        set(uuid)
+
+        return uuid
+    }
+
+    /// Explicitly set an existing UUID value
+    ///
+    /// - parameters:
+    ///     * value: A valid `UUID` to set the metric to
+    public func set(_ value: UUID) {
+        guard !self.disabled else { return }
+
+        Dispatchers.shared.launchAPI {
+            glean_uuid_set(Glean.shared.handle, self.handle, value.uuidString.lowercased())
+        }
+    }
+
+    /// Tests whether a value is stored for the metric for testing purposes only. This function will
+    /// attempt to await the last task (if any) writing to the the metric's storage engine before
+    /// returning a value.
+    ///
+    /// - parameters:
+    ///     * pingName: represents the name of the ping to retrieve the metric for.
+    ///                 Defaults to the first value in `sendInPings`.
+    /// - returns: true if metric value exists, otherwise false
+    func testHasValue(_ pingName: String? = nil) -> Bool {
+        Dispatchers.shared.assertInTestingMode()
+
+        let pingName = pingName ?? self.sendInPings[0]
+        return glean_uuid_test_has_value(Glean.shared.handle, self.handle, pingName) != 0
+    }
+
+    /// Returns the stored value for testing purposes only. This function will attempt to await the
+    /// last task (if any) writing to the the metric's storage engine before returning a value.
+    ///
+    /// Throws a `String` exception if no value is stored
```

Comment:
Does it still throw a `String` exception?

Reply:
In `Timespan` I changed that sentence to:

```
/// Throws a "Missing value" exception if no value is stored
```